File: src/test/java/instruments/ObjectMapperWrapper.java
package instruments;

import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.module.SimpleModule;

import java.io.File;

public class ObjectMapperWrapper {
    private final ObjectMapper objectMapper;

    public ObjectMapperWrapper(ObjectMapper objectMapper) {
        this.objectMapper = objectMapper;
    }

    public <T> T readValue(File file, Class<T> valueType) {
        try {
            return objectMapper.readValue(file, valueType);
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }

    public <T> T readValue(String content, Class<T> valueType) {
        try {
            return objectMapper.readValue(content, valueType);
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }

    public <T> T readValue(String content, TypeReference<T> valueType) {
        try {
            return objectMapper.readValue(content, valueType);
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }

    public String writeValueAsString(Object value) {
        try {
            return objectMapper.writeValueAsString(value);
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }

    public void findAndRegisterModules() {
        objectMapper.findAndRegisterModules();
    }

    public void registerModule(SimpleModule simpleModule) {
        objectMapper.registerModule(simpleModule);
    }

    public void registerModules(SimpleModule... simpleModule) {
        objectMapper.registerModules(simpleModule);
    }

    public byte[] writeValueAsBytes(Object object) {
        try {
            return objectMapper.writeValueAsBytes(object);
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }
}

File: src/test/kotlin/instruments/ObjectMapperWrapper.kt
package instruments;

import com.fasterxml.jackson.core.type.TypeReference
import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.databind.module.SimpleModule
import org.springframework.boot.test.context.TestComponent
import java.io.File

@TestComponent
class ObjectMapperWrapper(val objectMapper: ObjectMapper) {
    fun <T> readValue(file: File?, valueType: Class<T>?): T = suppressChecked {
        objectMapper.readValue(file, valueType)
    }

    fun <T> readValue(content: String?, valueType: Class<T>?): T = suppressChecked {
        objectMapper.readValue(content, valueType)
    }

    fun <T> readValue(content: String?, valueType: TypeReference<T>?): T = suppressChecked {
        objectMapper.readValue(content, valueType)
    }

    fun writeValueAsString(value: Any?): String? = suppressChecked { objectMapper.writeValueAsString(value) }

    fun findAndRegisterModules() =
        objectMapper.findAndRegisterModules()

    fun registerModule(simpleModule: SimpleModule?) =
        objectMapper.registerModule(simpleModule)

    fun registerModules(vararg simpleModule: SimpleModule?) =
        objectMapper.registerModules(*simpleModule)

    fun writeValueAsBytes(`object`: Any?): ByteArray? = suppressChecked {
        objectMapper.writeValueAsBytes(`object`)
    }

    companion object {
        inline fun <T> suppressChecked(f: () -> T) = try {
            f()
        } catch (e: Exception) {
            throw RuntimeException(e)
        }
    }
}


